### PR TITLE
possibility to publish the plugin locally, for dev purposes

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

**Comment by wind57** [comment]
I am opening this PR, but let's talk about it too and see your thoughts also.

So currently to test a plugin change - you need these changes in gradle build file and issue a simple `gradle clean && gradle publishToMavenLocal`. Should we have these here and may be add a small documentation note on how to build this if you want to contribute?

**Comment by jekkel** [comment]
I agree it should be easy to publish to maven local for dev purposes and documenting this in [CONTRIBUTION.md](https://github.com/kiwigrid/helm-gradle-plugin/blob/master/CONTRIBUTING.md) would be great :+1: 

But from my understanding adding the `maven-publish` plugin is all that's needed for this, the changes to `repositories` and `publishing` are obsolete, aren't they?

However, as noted in the [further work list](https://github.com/kiwigrid/helm-gradle-plugin/blob/master/README.md#further-work) we urgently need some tests as sporadic developer tests are not sustainable. Would you be able to draft testing according to [gradle plugin development documentation](https://docs.gradle.org/current/userguide/custom_plugins.html#sec:writing_tests_for_your_plugin)?

**Comment by wind57** [comment]
you are right, only that plugin is needed.

My main priority right now is to get this working, for all of our uses cases : otherwise I simply can't "sell this" to my team.  As soon as that is done, I hope some of my teammates will jump into helping out here too and provide tests. But even if that does not happen - I will do it at some point in time. When it reaches our production code base, there is simply no other way but for me to get involved and support this; thus proper unit tests. 

I hope I make sense.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,6 +6,25 @@ What to contribute
 
 Pick a [ticket](https://github.com/kiwigrid/helm-gradle-plugin/issues), or an item off the [future work list](README.md#further-work) in the or just the one feature you want to see here.
 
+Build plugin locally
+---
+
+To build and test locally, simply issue : 
+
+```
+gradle clean && gradle publishToMavenLocal
```
Comment (kiwi-oss) [inline CONTRIBUTING.md:15]: Is there a particular reason for calling Gradle twice?
```suggestion
gradle clean publishToMavenLocal
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,6 +6,25 @@ What to contribute
...
+```
+
+you can optionally also change the version to be published in your local maven repo by changing `gradle.properties` file.
```
Comment (kiwi-oss) [inline CONTRIBUTING.md:18]: Or by setting the property from the command line:
```shell
gradle publishToMavenLocal --project-prop version=1.2.3
```

Comment (wind57) [reply CONTRIBUTING.md:18]: nice. thank you, I was not aware you could do this

File: CONTRIBUTING.md (hunk 1 above)
Comment (wind57) [reply CONTRIBUTING.md]: just an old habit from other terminal commands. you are right, edited.

File: CONTRIBUTING.md (hunk 2 above)
Comment (kiwi-oss) [reply CONTRIBUTING.md:18]: [Here you can read more about it](https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_system_properties) if you're interested.

**Comment by wind57** [comment]
@jekkel this seems to be ready to go too? no rush, just making sure. otherwise, thank you @kiwi-oss for the comments.

**Comment by jekkel** [comment]
@kiwi-oss could you please re-review?
@wind57 could you remove the "draft" status from the PR if it's good to go from your side?

**Comment by wind57** [comment]
@jekkel done.

**Comment by jekkel** [comment]
I just noticed the branch is not up to date with the base branch. Could you please rebase? If you like you can also squash or reword your commits while doing this. We like to have meaningful commit messages and commits which contain related changes (take this as a guideline, not a law).

**Review by kiwi-oss — APPROVED** [review]

**Pull request merged**